File: backend/orchestrator-prompt/system/judge_idea.py
# アイデア評価
def get_judge_idea_system_message(
    ideation_function: str,
    ideation_use: str,
    ideation_market: str,
    ideation_country: str,
) -> str:
    JUDGE_IDEATION_SYSTEM_MESSAGE = f"""\
    ### 役割
    あなたは製造業の法律や規則に精通した弁護士です。

    ### 指示
    以下の#入力欄について、製造業に関連する法律や規制に違反していないかを評価してください。
    違反している場合は以下の#出力形式に従って結果を出力してください。

    ### 出力形式
    違反1
    違反箇所：
    基準となる法令：
    コメント：

    ### 入力欄
    新製品のアイデアの具体的な機能：{ideation_function}
    新製品のアイデアの用途：{ideation_use}
    新製品のアイデアの対象市場：{ideation_market}
    法的評価を行いたい国や地域：{ideation_country}
    """
    return JUDGE_IDEATION_SYSTEM_MESSAGE


def get_judge_idea_message(
    ideation_function: str,
    ideation_use: str,
    ideation_market: str,
    ideation_country: str,
):
    messages = [
        {
            "role": "user",
            "content": get_judge_idea_system_message(
                ideation_function,
                ideation_use,
                ideation_market,
                ideation_country,
            ),
        },
    ]
    return messages

File: frontend/src/app/(dashboard)/compliance-judge/_actions/judgeIdea.ts
'use server';

import { judgeIdeaDB } from '@/app/_db/judge';
import { LLMserviceBackEndLog } from '@/app/_types/logs/llm-services';
import { getCurrentUser } from '@/app/_utils/auth';
import { getMessage } from '@/app/_utils/message';
import { useCaseAzureFunctions } from '../../../../../azure-functions';
import { judgeIdeaContainer } from '../../../../../cosmos';

type IdeaResponse = {
  answer: string;
  log: LLMserviceBackEndLog<'judge'>;
};

type IdeaErrorResponse = {
  error: string;
};

export async function judgeIdea(
  id: string,
  ideationFunction: string,
  ideationUse: string,
  ideationMarket: string,
  ideationCountry: string
): Promise<IdeaResponse | IdeaErrorResponse> {
  const user = await getCurrentUser();
  try {
    const response = await useCaseAzureFunctions.sendJson<
      {
        ideationFunction: string;
        ideationUse: string;
        ideationMarket: string;
        ideationCountry: string;
      },
      IdeaResponse
    >('judge-idea', 'POST', {
      ideationFunction,
      ideationUse,
      ideationMarket,
      ideationCountry,
    });

    // log
    await judgeIdeaDB(judgeIdeaContainer, {
      id,
      userId: user.id,
      useName: user.name,
      userEmail: user.email,
      userDepartmentName: user.departmentName,
      createdAt: new Date(),
      ideationFunction,
      ideationUse,
      ideationMarket,
      ideationCountry,
      outputForm: response.answer,
      log: response.log,
    });

    return {
      answer: response.answer,
      log: response.log,
    };
  } catch (error) {
    console.error('Create idea error:', error);
    return { error: error instanceof Error ? error.message : getMessage('E_F_00110', '作成結果') };
  }
}
